src/main/java/chess/Rook.java

```diff
@@ -1,33 +1,29 @@
 package chess;
 
 import java.util.ArrayList;
-import java.util.List;
-
 class Rook extends Piece implements ICastling {
-    Rook(final Color color, final Position source) {
+
+    private GetPossibleMovePiece getPossibleMovePiece;
+    Rook(final Color color, final Position source, final ChessBoard chessBoard) {
         super.setColor(color);
         super.setSymbol(Symbol.R);
         super.setStatus(Status.ALIVE);
         super.setPosition(source);
+        super.setChessBoard(chessBoard);
+        this.getPossibleMovePiece = new GetPossibleMovePiece(this);
     }
 
-    public void kill(final Piece target) {
+    public ArrayList<Position> getPossibleMoves(final ChessBoard chb) {
+        ArrayList<Position> arraypositions = new ArrayList<Position>();
+        int row = this.getPosition().getRow();
+        int column = this.getPosition().getColumn();
+        getPossibleMovePiece.diagonal(arraypositions, column + 1, row, 0, +1);
```

Comment:
Please, don't forget to change the name of "diagonal" method to be more descriptive and accesible to Rook and Queen.

Reply:
Change made can you review it? please